### PR TITLE
Dashboard revamp

### PR DESCRIPTION
**Pull request by eugenepeh**
This PR contains the work done by @ongspxm 

Proposed merge message:
```
There are 2 main templates loaded separately to process the information
from the generated report. This requires the same template files to be
regenerated multiple times, which greatly increases the size of the
generated report unnecessarily. Ideally, it should only contains the
data for the report.

Also, as the dashboard uses vue for its templating engine mostly, let's
better utilize the framework to restructure the dashboard into a single
page application which loads content from the report dynamically.
```

### CODE REVIEW

**Comment by ongspxm** [comment]
@eugenepeh ready for merge

**Comment by eugenepeh** [comment]
@ongspxm please proposed a merge message for this PR

**Comment by ongspxm** [comment]
@damithc ready for review

**proposed commit message**

```
There are 2 main templates loaded seperately to process the information from the generated report. This requires the same template files to be regenerated multiple times, which greatly increases the size of the generated report unnecessarily, which ideally, should only contains the data for the report.

The current dashboard uses vue for its templating engine mostly. Let's more fully ultilize the vue framework to restructure the dashboard into a SPA which loads the content from the report dynamically.
```

**Comment by damithc** [comment]
`more fully ultilize` -> `better utilize`?
What's `SPA`? Use the full version of the term?

File: template/static/js/api.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/* util funcs */
```
Comment (damithc) [inline template/static/js/api.js]: Use full words e.g., functions

Comment (ongspxm) [reply template/static/js/api.js]: so i make a new pr to frontend for the nits?

**Comment by ongspxm** [comment]
**edited proposed message**

```
There are 2 main templates loaded seperately to process the information
from the generated report. This requires the same template files to be
regenerated multiple times, which greatly increases the size of the
generated report unnecessarily, which ideally, should only contains the
data for the report.

The current dashboard uses vue for its templating engine mostly. Let's
more better ultilize the vue framework to restructure the dashboard into
a single page application which loads the content from the report dynamically.
```

File: template/static/js/api.js (hunk 1 above)
Comment (damithc) [reply template/static/js/api.js]: Nits can be fixed in this PR. No need to get my review again though.

**Comment by ongspxm** [comment]
nits and conflict resolved in #173

**Comment by eugenepeh** [comment]
@ongspxm seems like the additional instructions on `UserGuide` were not removed, please do a proper merge to the master to resolve the conflict.

**Comment by ongspxm** [comment]
roger that, opened a new PR in #176

**Comment by eugenepeh** [comment]
@ongspxm seems like need to merge one more time due to the #175 quick fix

**Comment by eugenepeh** [comment]
Also for the PR title, since there were many changes, I will just simplify it to `Dashboard revamp`.
Do you think that's appropriate? @ongspxm @damithc

**Comment by damithc** [comment]
> I will just simplify it to Dashboard revamp.
> Do you think that's appropriate?

Sure

**Pull request merged**